authui/src/getColorScheme.ts

```diff
@@ -0,0 +1,22 @@
+export function getColorScheme(): string {
+  const queryResult = window.matchMedia("(prefers-color-scheme: dark)");
+  const htmlElement = document.documentElement;
+  const darkThemeEnabled =
+    htmlElement.getAttribute("data-dark-theme-enabled") === "true";
+
+  let explicitColorScheme = "";
+  const metaElement = document.querySelector("meta[name=x-color-scheme]");
+  if (metaElement instanceof HTMLMetaElement) {
+    explicitColorScheme = metaElement.content;
+  }
+
+  const implicitColorScheme = queryResult.matches ? "dark" : "light";
+
+  const colorScheme = !darkThemeEnabled
+    ? "light"
+    : explicitColorScheme !== ""
+      ? explicitColorScheme
+      : implicitColorScheme;
+
+  return colorScheme;
+}
```

Comment:
This code duplicates with colorscheme.ts. Due to the nature of colorscheme.ts, it cannot import other modules. So please a comment in this file, and in colorscheme.ts, saying that the code is duplicate and ask the future editor to keep the two synced.

Reply:
colorscheme.ts has been changed. Please update.